pkg/agent/controller/bgp/controller_test.go

```diff
@@ -2161,3 +2161,72 @@ func TestGetBGPPolicyInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestGetBGPPeerStatus(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		existingState         *bgpPolicyState
+		expectedCalls         func(mockBGPServer *bgptest.MockInterfaceMockRecorder)
+		expectedBgpPeerStatus []bgp.PeerStatus
+		expectedErr           string
+	}{
+		{
+			name:          "bgpPolicyState exists",
+			existingState: &bgpPolicyState{},
+			expectedCalls: func(mockBGPServer *bgptest.MockInterfaceMockRecorder) {
+				mockBGPServer.GetPeers(context.Background()).Return([]bgp.PeerStatus{
```

Comment:
nit: probably less verbose / more readable to defined `ctx := context.Background()` once at the beginning of the `TestGetBGPPeerStatus` function.